index.html

```diff
@@ -146,125 +149,353 @@ <h2>
+          "event">pointerlockchange</dfn> and <dfn class=
+          "event">pointerlockerror</dfn>. Refer to algorithm of
+          [[[#requestPointerLock]]] for detail.
+        </p>
+        <p class="note">
+          Magnification software increases the size of content on the screen.
+          It uses the mouse to move the magnified point of focus around. When a
+          pointer lock is initiated, the magnification software needs to switch
+          to using the keyboard to move the magnified point of focus around
+          instead. When a {{pointerlockchange}} event is fired, web browsers
+          therefore need to make sure the event is communicated to assistive
+          technologies like screen magnifiers.
+        </p>
+      </section>
+      <section>
+        <h3>
+          <dfn>Exit Pointer Lock</dfn>
+        </h3>
+        <p>
+          The process of exiting pointer lock, given an |element:Element|, is
+          as follows:
+        </p>
+        <ol class="algorithm">
+          <li>The system mouse cursor must be displayed again and positioned at
+          [=cursor position=].
+          </li>
+          <li>[=Queue an element task=] on the [=user interaction task source=]
+          to [=fire an event=] named {{pointerlockchange}} at the given
+          |element:Element|'s [=Node/node document=].
+          </li>
+          <li>Exit the [=pointer lock state=] by setting [=pointer-lock
+          target=], [=pointer-lock options=], and [=cursor position=] to null.
+          </li>
+        </ol>
+      </section>
     </section>
-    <section data-dfn-for="Element" data-link-for="Element">
+    <section data-dfn-for="Element" data-link-for="Element" id=
+    "requestPointerLock">
       <h2>
         Extensions to the `Element` Interface
       </h2>
       <p>
-        The <a>Element</a> interface is extended to provide the ability to
-        request the the pointer be locked.
+        The {{Element}} interface is extended to provide the ability to request
+        the pointer be locked.
       </p>
       <pre class="idl">
         partial interface Element {
-          undefined requestPointerLock();
+          Promise&lt;undefined&gt; requestPointerLock(optional PointerLockOptions options = {});
         };
       </pre>
+      <pre class='example'>
+        const lock_element = document.getElementById("lock_element");
+        const lock_button = document.getElementById("lock");
+        lock_button.addEventListener("click", async (event) =&gt; {
+            try {
+                await lock_element.requestPointerLock({ unadjustedMovement: true });
+                console.log("successfully locked!");
+            } catch (e) {
+                console.log("lock failed with error: ", e);
+            }
+        });
+    </pre>
       <dl>
         <dt>
-          <dfn>requestPointerLock()</dfn> method
+          requestPointerLock({{PointerLockOptions}} options) method
```

Comment:
Hi @scheib / @mustaqahmed  / @marcoscaceres  / @james-howard,
I've updated the algorithm to use a `parallel queue` enabling it to handle consecutive requests.
Failure condition checks are now performed before enqueuing steps. 
For unresolved discussions, I've added relevant issue reference in-place. 
Please help to review it.
Thanks!
Alvin

Reply:
Hi @scheib / @mustaqahmed / @marcoscaceres, 
Do you get a chance to review this? 
Thanks!
Alvin

Reply:
Thanks @scheib / @mustaqahmed  / @james-howard for reviewing this. 

Hi @marcoscaceres , 
Do you get a chance to review this?
Thanks!
Alvin